src/layered_config_tree/main.py

```diff
@@ -640,7 +678,10 @@ def __getattr__(self, name: str) -> Any:
                 "access (i.e. dot notation). Use dictionary access instead "
                 "(i.e. bracket notation)."
             )
-        return self.get_from_layer(name)
+        if name not in self:
```

Comment:
`.get()` is now called below instead of `get_from_layer()`. But since `get()` defaults to returning a default of None, we need to raise here if you're trying to access a non-existent value from dot-notation or index-notation.

Reply:
Why is that better than both calling `.get()`?

Reply:
Oh are you saying to move the `if name not in self` logic into one of the two and then have the other just call that? That feels weird to me for some reason but I do like the DRYness.